core/schema/host.go

```diff
@@ -207,20 +211,43 @@ type hostSocketArgs struct {
 	Path string
 }
 
-func (s *hostSchema) socket(ctx context.Context, host *core.Host, args hostSocketArgs) (*core.Socket, error) {
-	clientMetadata, err := engine.ClientMetadataFromContext(ctx)
+func (s *hostSchema) socket(ctx context.Context, host *core.Host, args hostSocketArgs) (inst dagql.Instance[*core.Socket], err error) {
+	socketStore, err := host.Query.Sockets(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get client metadata: %w", err)
+		return inst, fmt.Errorf("failed to get socket store: %w", err)
 	}
-	mainClientCallerID, err := host.Query.MainClientCallerID(ctx)
+
+	accessor, err := core.GetClientResourceAccessor(ctx, host.Query, args.Path)
```

Comment:
Nonblocking: I guess it's possible for two different types of objects to have the same accessor, considering this is basically `currentModule.Source.ID:/foo` - but it's not likely for that to be an issue, since those would be interacting with different subsystems (e.g. socket LLB ID, secret LLB ID), and these keys are only relevant within their subsystem.

Reply:
True, that would be possible but yeah I think that should be okay in the current state of things